Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -359,72 +353,124 @@ - (void)configureViews
         }
     }
 
-    UIStackView *_Nullable textStackView = nil;
-
     // We render malformed messages as "empty text" messages,
     // so create a text view if there is no body media view.
     if (self.hasBodyText || !bodyMediaView) {
-        OWSMessageTextView *_Nullable bodyTextView = nil;
-        bodyTextView = [self configureBodyTextView];
-
-        textStackView = [UIStackView new];
-        textStackView.axis = UILayoutConstraintAxisVertical;
-        textStackView.alignment = UIStackViewAlignmentFill;
-        // TODO: Review
-        textStackView.spacing = self.textViewVSpacing;
-        textStackView.layoutMarginsRelativeArrangement = YES;
-        textStackView.layoutMargins = UIEdgeInsetsMake(self.conversationStyle.textInsetTop,
-            self.conversationStyle.textInsetHorizontal,
-            self.conversationStyle.textInsetBottom,
-            self.conversationStyle.textInsetHorizontal);
-        [self.stackView addArrangedSubview:textStackView];
-        [textStackView addArrangedSubview:bodyTextView];
+        [self configureBodyTextView];
+        [textViews addObject:self.bodyTextView];
 
         [self.viewConstraints addObjectsFromArray:@[
-            [bodyTextView autoSetDimension:ALDimensionHeight toSize:bodyTextContentSize.height],
+            [self.bodyTextView autoSetDimension:ALDimensionHeight toSize:bodyTextContentSize.height],
         ]];
 
         UIView *_Nullable tapForMoreLabel = [self createTapForMoreLabelIfNecessary];
         if (tapForMoreLabel) {
-            [textStackView addArrangedSubview:tapForMoreLabel];
+            [textViews addObject:tapForMoreLabel];
             [self.viewConstraints addObjectsFromArray:@[
                 [tapForMoreLabel autoSetDimension:ALDimensionHeight toSize:self.tapForMoreHeight],
             ]];
         }
     }
 
-    OWSMessageFooterView *footerView = self.footerView;
-    [footerView configureWithConversationViewItem:self.viewItem];
-    if (textStackView) {
-        // Display footer below text.
-        [textStackView addArrangedSubview:self.footerView];
-        [self.footerView setHasShadows:NO viewItem:self.viewItem];
-    } else if (bodyMediaView) {
+    if (self.viewItem.shouldHideFooter) {
+        // Do nothing.
+    } else if (shouldFooterOverlayMedia) {
+        OWSAssert(bodyMediaView);
         // Display footer over media.
```

Comment:
nit: the code is pretty literate, not sure these comments add anything.

Reply:
Removed.